sycl/doc/design/CompileTimeProperties.md

```diff
@@ -596,6 +582,56 @@ we do not allow non-fundamental types, how do we represent properties like
 allow `std::tuple`, where the type of each element is one of the fundamental
 types listed above.
 
+### IR representation via `@llvm.global.annotations`
+
+Properties that are implemented using
+`[[__sycl_detail__::add_ir_attributes_global_variable()]]`, are represented in LLVM IR
+as elements in the global variable `@llvm.global.annotations`. `@llvm.global.annotations` is an array of `{ ptr, ptr, ptr, i32, ptr}` structs, where each struct represents an annotation on a global pointer. The values of each field of the struct are as follows:
```

Comment:
You are right that this is not documented in the LangRef, but are wrong that this is our invention. It has actually been in clang since at least clang 6.0.0 (you can see an example of this here: https://godbolt.org/z/eqz114o8G). A bug has been filed to LLVM to document the feature (https://github.com/llvm/llvm-project/issues/66004).